providers/dynamodb/shedlock-provider-dynamodb/src/main/java/net/javacrumbs/shedlock/provider/dynamodb/DynamodbLockProvider.java

```diff
@@ -0,0 +1,306 @@
+    static final String ID = "_id";
+    static final String NAME = "name";
+
+
+    /**
+     * Uses Amazon Dynamo Db to coordinate locks
+     *
+     * @param amazonDynamoDB Amazon Dynamo DB Database.
+     *                       Sample code :
+     *                       <pre>
+     *                                           AmazonDynamoDB amazonDynamoDB = AmazonDynamoDBClientBuilder.standard()
+     *                                           .withEndpointConfiguration(
+     *                                                new AwsClientBuilder.EndpointConfiguration(AMAZON_DYNAMODB_URL, REGION_US_WEST_2))
+     *                                                .build();
+     *                                           </pre>
+     */
+    public DynamodbLockProvider(AmazonDynamoDB amazonDynamoDB) {
+        this(amazonDynamoDB, "shedLock", Duration.of(1, ChronoUnit.MINUTES), 1);
+    }
+
+    /**
+     * Uses Mongo to coordinate locks
+     *
+     * @param amazonDynamoDB         Amazon Dynamo DB Database.
+     *                               Sample code :
+     *                               <pre>
+     *                                                   AmazonDynamoDB amazonDynamoDB = AmazonDynamoDBClientBuilder.standard()
+     *                                                   .withEndpointConfiguration(
+     *                                                        new AwsClientBuilder.EndpointConfiguration(AMAZON_DYNAMODB_URL, REGION_US_WEST_2))
+     *                                                        .build();
+     *                                                   </pre>
+     * @param tableName              collection to store the locks
+     * @param minLockTime            minimum locking duration for a process. This is used to determine the required throughput
+     *                               for dynamo db. Use this as tuning parameter to optimize the cost and avoid any throttling
+     *                               by dynamo db. Higher minLockTime will reduce cost and also reduce throughput of dynamo db
+     * @param concurrentProcessCount number of concurrent nodes/processes running at a given time. This is used to
+     *                               determine the required throughput for dynamo db. Use this as tuning parameter
+     *                               to optimize the cost and avoid any throttling by dynamo db. Higher value
+     *                               will increase the cost and also increase throughput of dynamo db
+     */
+    public DynamodbLockProvider(AmazonDynamoDB amazonDynamoDB, String tableName, Duration minLockTime, long concurrentProcessCount) {
+        this(new DynamodbAccessor(amazonDynamoDB, tableName, minLockTime, concurrentProcessCount));
+    }
+
+    DynamodbLockProvider(DynamodbAccessor accessor) {
+        super(accessor);
+    }
+
+    public static class DynamodbAccessor extends AbstractStorageAccessor {
+        private boolean IS_TABLE_EXIST = false;
```

Comment:
Why is this in capital letters?

Reply:
This is a little remnant from my previous effort when this wasn't part of the constructor. I believe this should not be required any more.